open-sphere-plugins/feature-actions/src/main/java/io/opensphere/featureactions/controller/FeatureActionStateController.java

```diff
@@ -0,0 +1,260 @@
+    }
+
+    @Override
+    public void activateState(String id, String description, Collection<? extends String> tags, Node node)
+    {
+        try
+        {
+            NodeList featureActionNodes = StateXML.getChildNodes(node, "/:state/:featureActions/:featureAction");
+            for (int i = 0; i < featureActionNodes.getLength(); i++)
+            {
+                FeatureAction newFeatureAction = new FeatureAction();
+                Node featureActionNode = featureActionNodes.item(i);
+                try
+                {
+                    newFeatureAction = XMLUtilities.readXMLObject(featureActionNode, FeatureAction.class);
+                }
+                catch (JAXBException e)
+                {
+                    LOGGER.error(e.getMessage(), e);
+                }
+                String layerPath = StateXML.newXPath().evaluate("/" + ModuleStateController.STATE_QNAME
+                        + "/:featureActions/:featureAction[@type=\"type\"]", featureActionNode);
+                myRegistry.add(layerPath, Collections.singleton(newFeatureAction), this);
+                addResource(id, new Pair<>(newFeatureAction, layerPath));
+            }
+        }
+        catch (XPathExpressionException e)
+        {
+            LOGGER.error(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void activateState(String id, String description, Collection<? extends String> tags, StateType state)
+            throws InterruptedException
+    {
+        FeatureActionArrayType featureActionList = StateUtilities.getFeatureActions(state);
+        for (FeatureActionType featureActionType : featureActionList.getFeatureAction())
+        {
+            FeatureAction newFeatureAction = new FeatureAction();
+            newFeatureAction.setName(featureActionType.getTitle() + " - " + id);
+            newFeatureAction.setEnabled(featureActionType.isActive());
+            newFeatureAction.setGroupName(featureActionType.getDescription());
+            WFS100FilterToDataFilterConverter filterConverter = new WFS100FilterToDataFilterConverter();
+            newFeatureAction.setFilter(filterConverter.apply(featureActionType.getFilter()));
+            StyleAction style = new StyleAction();
+            FeatureStyleActionType act = (FeatureStyleActionType)featureActionType.getActions().getAction().get(0).getValue();
+            style.getStyleOptions().setColor(ColorUtilities.convertFromHexString(act.getColor(), 1, 2, 3, 0));
+            style.getStyleOptions().setSize(act.getSize().intValue());
+            newFeatureAction.getActions().add(style);
+            myRegistry.add(featureActionType.getType(), Collections.singleton(newFeatureAction), this);
+            addResource(id, new Pair<>(newFeatureAction, featureActionType.getType()));
+        }
+    }
+
+    @Override
+    public boolean canActivateState(Node node)
+    {
+        return StateXML.anyMatch(node, "/:state/:featureActions/:featureAction");
```

Comment:
I would recommend making a FeatureActionStateConstants class with these string values (similar to CSVStateConstants, StateConstants, etc.)